basics/basic/src/main/java/org/jvnet/jaxb2_commons/plugin/namespace_prefix/NamespacePrefixPlugin.java

```diff
@@ -204,12 +205,11 @@ private static void addNamespacePrefix(JClass xmlNsClass, JAnnotationArrayMember
         ns.param("prefix", prefix);
     }
 
-    @SuppressWarnings("unchecked")
     private static JAnnotationUse getOrAddXmlSchemaAnnotation(JPackage p, JClass xmlSchemaClass) {
 
         JAnnotationUse xmlAnn = null;
 
-        final List<JAnnotationUse> annotations = getAnnotations(p);
+        final Collection<JAnnotationUse> annotations = getAnnotations(p);
```

Comment:
There's a change in `getAnnotations` function to avoid reflective access to property which is know accessible via java API. But the API returns a Collection here, so to avoid unnecessary cast or conversion, I switched back to Collection.

```
    private static Collection<JAnnotationUse> getAnnotations(JPackage p) {
        return p.annotations();
    }
```